src/NUnitTestAdapter/NUnit3TestDiscoverer.cs

```diff
@@ -43,76 +43,69 @@ public void DiscoverTests(IEnumerable<string> sources, IDiscoveryContext discove
-                    {
-                        // we skip the native c++ binaries that we don't support.
-                        TestLog.AssemblyNotSupportedWarning(sourceAssembly);
-                    }
-                    catch (FileNotFoundException ex)
+                    else
                     {
-                        // Either the NUnit framework was not referenced by the test assembly
-                        // or some other error occured. Not a problem if not an NUnit assembly.
-                        TestLog.DependentAssemblyNotFoundWarning(ex.FileName, sourceAssembly);
-                    }
-                    catch (FileLoadException ex)
-                    {
-                        // Attempts to load an invalid assembly, or an assembly with missing dependencies
-                        TestLog.LoadingAssemblyFailedWarning(ex.FileName, sourceAssembly);
-                    }
-                    catch (TypeLoadException ex)
-                    {
-                        if (ex.TypeName == "NUnit.Framework.Api.FrameworkController")
-                            TestLog.SendWarningMessage("   Skipping NUnit 2.x test assembly");
+                        var msgNode = loadResult.SelectSingleNode("properties/property[@name='_SKIPREASON']");
+                        if (msgNode != null && (new[] { "contains no tests", "Has no TestFixtures" }).Any(msgNode.GetAttribute("value").Contains))
+                            TestLog.SendInformationalMessage("Assembly contains no NUnit 3.0 tests: " + sourceAssembly);
                         else
-                            TestLog.SendErrorMessage("Exception thrown discovering tests in " + sourceAssembly, ex);
+                            TestLog.NUnitLoadError(sourceAssembly);
                     }
-                    catch (Exception ex)
-                    {
+                }
+                catch (BadImageFormatException)
+                {
+                    // we skip the native c++ binaries that we don't support.
+                    TestLog.AssemblyNotSupportedWarning(sourceAssembly);
+                }
+                catch (FileNotFoundException ex)
+                {
+                    // Either the NUnit framework was not referenced by the test assembly
+                    // or some other error occured. Not a problem if not an NUnit assembly.
+                    TestLog.DependentAssemblyNotFoundWarning(ex.FileName, sourceAssembly);
+                }
+                catch (FileLoadException ex)
+                {
+                    // Attempts to load an invalid assembly, or an assembly with missing dependencies
+                    TestLog.LoadingAssemblyFailedWarning(ex.FileName, sourceAssembly);
+                }
+                catch (TypeLoadException ex)
+                {
+                    if (ex.TypeName == "NUnit.Framework.Api.FrameworkController")
+                        TestLog.SendWarningMessage("   Skipping NUnit 2.x test assembly");
```

Comment:
It's not a line I really changed, though hard to tell from the formatting. In my mind it's a warning if you thought the adapter would pick it up. Since we can't read minds, I guess it's a warning only if the other adapter is not present - but we can't figure that out easily either, and probably shouldn't.

I'd say it's a warning that we may degrade to informational in the future.